lib/lightning_web/live/components/user_deletion_modal.ex

```diff
@@ -42,15 +42,22 @@ defmodule LightningWeb.Components.UserDeletionModal do
         } = _user_params,
         socket
       ) do
-    case Accounts.schedule_user_deletion(socket.assigns.user, email) do
-      {:ok, _user} ->
-        {:noreply,
-         socket
-         |> put_flash(:info, "User scheduled for deletion")
-         |> logout_after_deletion()}
+    if @can_delete_account do
```

Comment:
I am not sure you can access this variable like this. You need to get it from `socket.assigns.can_delete_account` and even before that you may need to get it in the `update/2` function.